service/src/main/java/skills/services/ContactUsersService.groovy

```diff
@@ -75,10 +123,28 @@ class ContactUsersService {
                         ]
                 )
                 emailNotifier.sendNotification(request)
+            }
+
+            List<String> batch = []
+            userIds.forEach({
+                batch.add(it)
+                if (batch.size() == 150) {
```

Comment:
suggesting to make this batch size configurable; set it low enough in the tests to exercise sending multiple notifications